source/common/stats/allocator_impl.cc

```diff
@@ -412,14 +412,19 @@ void AllocatorImpl::forEachSinkedCounter(SizeFn f_size, StatFn<Counter> f_stat)
 }
 
 void AllocatorImpl::forEachSinkedGauge(SizeFn f_size, StatFn<Gauge> f_stat) const {
+  auto f_stat_to_use = [&f_stat](Gauge& gauge) {
+    if (!(gauge.hidden())) {
+      f_stat(gauge);
+    }
+  };
   if (sink_predicates_ != nullptr) {
     Thread::LockGuard lock(mutex_);
     f_size(sinked_gauges_.size());
     for (auto gauge : sinked_gauges_) {
-      f_stat(*gauge);
+      f_stat_to_use(*gauge);
```

Comment:
We should actually do the filtering on this path when populating sinked_gauges_. No need to filter every time this is called.

This will (a) be faster and (b) f_size will give you the correct count.

Reply:
@pradeepcrao should we consider *only* applying the 'hidden' filter for sinks if there is no sinked_predicates_ entry?

It strikes me now as a little odd that the sink-predicate interface is intended to delegate full control of what gets sinked to whoever writes the sink-predicate, and this would then apply a further filter beyond that.

I think in our use-case for sinked-predicates we simply wouldn't put the hidden gauges into the sink-predicate.

Reply:
I did think about that, and can be convinced either way. Now that I think about it some more, given the fact that not everyone probably uses (or can use) the admin interface for sinks, it might be beneficial to have a way to see hidden stats in sinks as well.

It makes sense to honour the sink predicates to show hidden stats if they return true for them (and it's trivial for sink predicates to return false for hidden stats).

Reply:
Done